addon/lib/connect.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import reduxStore from '../services/redux-store';
+
+const {
+  computed,
+  defineProperty,
+  inject: { service }
+} = Ember;
+
+export default function connect(mapStateToComputed, mapDispatchToActions, storeService) {
+  return function wrapWithConnect(WrappedComponent) {
+    return WrappedComponent.extend({
+      reduxStore: storeService || service('reduxStore'),
+
+      init() {
+        this._super(...arguments);
```

Comment:
`Component#init` doesn't return anything so I don't have to worry about that. However, it does do some internal setup however so it's important that it runs prior to any code here.